components/brave_vpn/brave_vpn_unittest.cc

```diff
@@ -808,6 +797,121 @@ TEST_F(BraveVPNServiceTest, LoadRegionDataFromPrefsTest) {
+  EXPECT_EQ(GetCurrentEnvironment(), skus::GetDefaultEnvironment());
+  EXPECT_FALSE(observer.GetPurchasedState().has_value());
+  // no order found for staging.
+  LoadPurchasedState(staging);
+  base::RunLoop().RunUntilIdle();
+  // The purchased state was not changed from dev env.
+  EXPECT_FALSE(observer.GetPurchasedState().has_value());
+  EXPECT_EQ(GetCurrentEnvironment(), skus::GetDefaultEnvironment());
+  EXPECT_EQ(GetPurchasedStateSync(), PurchasedState::PURCHASED);
+
+  observer.ResetStates();
+  staging = SetupTestingStoreForEnv(skus::kEnvStaging, false);
+  EXPECT_EQ(GetCurrentEnvironment(), skus::GetDefaultEnvironment());
+  EXPECT_FALSE(observer.GetPurchasedState().has_value());
+  // No region data for staging.
+  SetInterceptorResponse("");
+  LoadPurchasedState(staging);
+  base::RunLoop().RunUntilIdle();
+  // The purchased state was not changed from dev env.
+  EXPECT_FALSE(observer.GetPurchasedState().has_value());
+  EXPECT_EQ(GetCurrentEnvironment(), skus::GetDefaultEnvironment());
+  EXPECT_EQ(GetPurchasedStateSync(), PurchasedState::PURCHASED);
+
+  observer.ResetStates();
+  staging = SetupTestingStoreForEnv(skus::kEnvStaging, false);
+  EXPECT_EQ(GetCurrentEnvironment(), skus::GetDefaultEnvironment());
+  EXPECT_FALSE(observer.GetPurchasedState().has_value());
+  // Inactive staging subscription.
+  OnCredentialSummary(staging, R"({ "active": false } )");
+  // The purchased state was not changed from dev env.
+  EXPECT_FALSE(observer.GetPurchasedState().has_value());
+  EXPECT_EQ(GetCurrentEnvironment(), skus::GetDefaultEnvironment());
+  EXPECT_EQ(GetPurchasedStateSync(), PurchasedState::PURCHASED);
+
+  observer.ResetStates();
+  staging = SetupTestingStoreForEnv(skus::kEnvStaging, false);
+  EXPECT_EQ(GetCurrentEnvironment(), skus::GetDefaultEnvironment());
+  EXPECT_FALSE(observer.GetPurchasedState().has_value());
+  // Invalid staging subscription.
+  OnCredentialSummary(staging, R"([])");
+  // The purchased state was not changed from dev env.
+  EXPECT_FALSE(observer.GetPurchasedState().has_value());
+  EXPECT_EQ(GetCurrentEnvironment(), skus::GetDefaultEnvironment());
+  EXPECT_EQ(GetPurchasedStateSync(), PurchasedState::PURCHASED);
+}
+
+TEST_F(BraveVPNServiceTest, ResumeAfterSuspend) {
+  connection_state() = ConnectionState::CONNECTED;
+  needs_connect() = false;
+  Suspend();
+  EXPECT_TRUE(needs_connect());
+
+  connection_state() = ConnectionState::DISCONNECTED;
+  needs_connect() = false;
+  Suspend();
+  EXPECT_FALSE(needs_connect());
+}
+
+#endif
+
+TEST_F(BraveVPNServiceTest, CheckInitialPurchasedStateTest) {
+  // Purchased state is not checked for fresh user.
+  EXPECT_EQ(PurchasedState::NOT_PURCHASED, GetPurchasedStateSync());
+
+  // Dirty region list prefs to pretend it's already cached.
+  pref_service_.Set(prefs::kBraveVPNRegionList,
+                    base::Value(base::Value::Type::LIST));
+  ResetVpnService();
+  EXPECT_EQ(PurchasedState::LOADING, GetPurchasedStateSync());
+}
+
```

Comment:
I think this test also should be excluded from Android as `kBraveVPNRegionList` is only used on Desktop?

Reply:
done